apps/browser/src/autofill/services/insert-autofill-content.service.ts

```diff
@@ -48,13 +48,24 @@ class InsertAutofillContentService implements InsertAutofillContentServiceInterf
   }
 
   /**
-   * Determines if the page URL no longer matches one of the cipher's savedURL domains
-   * @param {string[] | null} savedUrls
+   * Determines if the page URL no longer matches one of the cipher's savedUrls or equivalentDomains
+   * @param {string[]} savedUrls
+   * @param {string[]} equivalentDomains
    * @returns {boolean}
    * @private
    */
-  private tabURLChanged(savedUrls?: AutofillScript["savedUrls"]): boolean {
-    return savedUrls && !savedUrls.some((url) => url.startsWith(window.location.origin));
+  private tabURLChanged(
+    savedUrls: AutofillScript["savedUrls"] = [],
+    equivalentDomains: AutofillScript["equivalentDomains"] = []
+  ): boolean {
+    let tabURLChanged = !savedUrls.some((url) => url.startsWith(window.location.origin));
+
+    // If the page origin doesn't match a savedURL, check the page location against equivalent domains
+    if (tabURLChanged) {
+      tabURLChanged = !equivalentDomains.some((url) => window.location.host.endsWith(url));
+    }
```

Comment:
```suggestion
    if (tabURLChanged) {
    const equivalentDomainsSet = new Set(equivalentDomains);
      tabURLChanged = !equivalentDomainsSet.has(window.location.host);
    }
```

I know that equivalentDomains is converted to an array of values in the background. If equivalentDomains is a simple set of domains... we likely could mitigate iterating over all of the elements by checking with `.has()`

Reply:
yeah, I was taking advantage of the imprecise matching, but per https://github.com/bitwarden/clients/pull/6355/files#r1332138443 we should probably match the logic that we use for url validation everywhere else